src/pudl/analysis/record_linkage/classify_plants_ferc1.py

```diff
@@ -0,0 +1,364 @@
+            columns=["construction_year"],
+        ),
+        "utility_id_ferc1": embed_dataframe.ColumnVectorizer(
+            transform_steps=[embed_dataframe.CategoricalVectorizer()],
+            columns=["utility_id_ferc1"],
+        ),
+        "fuel_fractions": embed_dataframe.ColumnVectorizer(
+            transform_steps=[
+                embed_dataframe.ColumnCleaner(cleaning_function="null_to_zero"),
+                embed_dataframe.NumericalVectorizer(),
+                embed_dataframe.NumericalNormalizer(),
+            ],
+            columns=_FUEL_COLS,
+        ),
+    }
+)
+
+
+@op
+def plants_steam_validate_ids(
+    ferc1_steam_df: pd.DataFrame, label_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Tests that plant_id_ferc1 times series includes one record per year.
+
+    Args:
+        ferc1_steam_df: A DataFrame of the data from the FERC 1 Steam table.
+
+    Returns:
+        The input dataframe, to enable method chaining.
+    """
+    # Add column of labels to steam df
+    ferc1_steam_df.loc[:, "plant_id_ferc1"] = label_df["record_label"]
+
+    ##########################################################################
+    # FERC PLANT ID ERROR CHECKING STUFF
+    ##########################################################################
+
+    # Test to make sure that we don't have any plant_id_ferc1 time series
+    # which include more than one record from a given year. Warn the user
+    # if we find such cases (which... we do, as of writing)
+    year_dupes = (
+        ferc1_steam_df.groupby(["plant_id_ferc1", "report_year"])["utility_id_ferc1"]
+        .count()
+        .reset_index()
+        .rename(columns={"utility_id_ferc1": "year_dupes"})
+        .query("year_dupes>1")
+    )
+    if len(year_dupes) > 0:
+        for dupe in year_dupes.itertuples():
+            logger.error(
+                f"Found report_year={dupe.report_year} "
+                f"{dupe.year_dupes} times in "
+                f"plant_id_ferc1={dupe.plant_id_ferc1}"
+            )
+    else:
+        logger.info("No duplicate years found in any plant_id_ferc1. Hooray!")
+
+    return ferc1_steam_df
+
+
+@op
+def merge_steam_fuel_dfs(
+    ferc1_steam_df: pd.DataFrame,
+    fuel_fractions: pd.DataFrame,
+) -> pd.DataFrame:
+    """Merge steam plants and fuel dfs to prepare inputs for ferc plant matching."""
+    ffc = list(fuel_fractions.filter(regex=".*_fraction_mmbtu$").columns)
+
+    # Grab fuel consumption proportions for use in assigning plant IDs:
+    return ferc1_steam_df.merge(
+        fuel_fractions[["utility_id_ferc1", "plant_name_ferc1", "report_year"] + ffc],
+        on=["utility_id_ferc1", "plant_name_ferc1", "report_year"],
+        how="left",
+    ).astype({"plant_type": str, "construction_type": str})
+
+
+@graph_asset
+def _out_ferc1__yearly_steam_plants_sched402_with_plant_ids(
+    core_ferc1__yearly_steam_plants_sched402: pd.DataFrame,
+    out_ferc1__yearly_steam_plants_fuel_by_plant_sched402: pd.DataFrame,
```

Comment:
I updated the name of this asset and the input assets after merging `dev` with the big PUDL rename